tracer.go

```diff
@@ -175,12 +175,16 @@ func (t *tracer) startSpanWithOptions(
 
 	var samplerTags []tag
 	var ctx SpanContext
-	if !hasParent {
+	debugRequest := (hasParent && parent.traceID == 0 && parent.debugID != "")
```

Comment:
yeah, encapsulating that logic in a function in SpanContext would be nice, could you also add a comment around it?